File: src/main/webapp/storage/src/app/_services/record.service.ts
import { Injectable } from '@angular/core';
import {HttpClient} from "@angular/common/http";
import {Record} from "../dto/Record";

const API_URL = 'http://localhost:8080/api/records';

@Injectable({
  providedIn: 'root'
})
export class RecordService {

  constructor(private http: HttpClient) { }

  getRecordsPage(pageNumber: number, pageSize: number, sort: string, way: string, id:string, code: string,
                 itemSize: string, pack: string, price: string, description: string, producer: string, count: string) {
    const params = {
      page: pageNumber,
      size: pageSize,
      sort: sort,
      way: way,
      id: id,
      code: code,
      itemSize: itemSize,
      pack: pack,
      price: price,
      description: description,
      producer: producer,
      count: count
    };
    return this.http.get<Record[]>(API_URL + '/page', {params});
  }

  getRecordsSize(id:string, code: string, itemSize: string, pack: string, price: string, description: string,
                 producer: string, count: string) {

    const params = {
      id: id,
      code: code,
      itemSize: itemSize,
      pack: pack,
      price: price,
      description: description,
      producer: producer,
      count: count
    };
    return this.http.get<number>(API_URL + '/size', {params});
  }
}

File: src/main/java/com/sonet/storage/service/RecordService.java
package com.sonet.storage.service;

import com.sonet.storage.dto.response.RecordResponse;
import com.sonet.storage.model.record.Record;
import com.sonet.storage.repository.RecordRepository;
import com.sonet.storage.specification.RecordSpecification;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.PageRequest;
import org.springframework.data.domain.Pageable;
import org.springframework.data.domain.Sort;
import org.springframework.stereotype.Service;

import java.util.List;
import java.util.stream.Collectors;

@Service
public class RecordService {

    @Autowired
    private RecordRepository recordRepository;

    public List<RecordResponse> getAllRecordsSize(String id, String code, String itemSize, String pack,
                                                  String price, String description, String producer,
                                                  String count) {

        List<Record> records = recordRepository.findAll(
                RecordSpecification.getRecordSpecification(id, code, itemSize, pack, price, description, producer, count));

        return records.stream().map(
                record -> new RecordResponse(
                        record.getId(),
                        record.getCount(),
                        record.getItem()
                )
        ).collect(Collectors.toList());
    }

    public List<RecordResponse> getRecordsPage(int page, int size, String sort, String way, String id, String code,
                                               String itemSize, String pack, String price, String description,
                                               String producer, String count) {
        Pageable pageable;
        if ("asc".equals(way)) {
            pageable = PageRequest.of(page, size, Sort.by(sort));
        } else if ("desc".equals(way)){
            pageable = PageRequest.of(page, size, Sort.by(sort).descending());
        } else {
            pageable = PageRequest.of(page, size);
        }

        Page<Record> records = recordRepository.findAll(RecordSpecification
                .getRecordSpecification(id, code, itemSize, pack, price, description, producer, count), pageable);

        return records.stream().map(
                record -> new RecordResponse(
                        record.getId(),
                        record.getCount(),
                        record.getItem()
                )
        ).collect(Collectors.toList());
    }
}
